### DISCUSSION TITLE
Does `brew leaves` take casks into account?

### DISCUSSION THREAD

**Discussion by jlledo** [Homebrew/discussions#722] [Casks] [answered]
I was looking for unused stuff and ran `brew leaves`, which showed me the `mono` formula. I was about to uninstall it, when I remembered it was a dependency of one of my installed casks, `godot-mono`.

I was a bit surprised by it, is this the expected behaviour?
Thanks for the help.

**Comment by carlocab** [comment, marked as answer]
>is this the expected behaviour?

Yes:

```
❯ brew help leaves
Usage: brew leaves

List installed formulae that are not dependencies of another installed formula.
```

However, I think `brew uninstall` would have stopped you from uninstalling `mono` because of the dependency. If it doesn't, that's a bug. PRs to improve `brew leaves` would be welcome.

**Reply by jlledo** [reply]
Thanks for the answer. I admit that I do get quite lost with regards to the terminology, I thought casks were formulae too.

Btw, I am able to uninstall `mono` while having `godot-mono` installed so I guess I'll file a bug report for that.

Re: the improvements, I'll open a suggestion issue for now since I don't know any Ruby. If I decide to try my hand in the future, is there anywhere I should ask for permission/feedback before opening the PR?

**Reply by carlocab** [reply]
Here's something that will help with the terminology: https://docs.brew.sh/Formula-Cookbook#homebrew-terminology

>is there anywhere I should ask for permission/feedback before opening the PR?

If the changes you're proposing aren't too big, you should just open a PR. If not, you can try asking here in the Discussions page for feedback (preferred), or to open an issue for a feature request at Homebrew/brew (it would be useful to make clear this a feature you are offering to implement).
